### PR TITLE
i18n: add script to detect and remove unused i18n keys

### LINKED ISSUE #354: Check for unused i18n keys
We need a script to automatically detect (and remove) unused i18n keys.  Consider using [i18n-unsued](https://www.npmjs.com/package/i18n-unused) to help with this.

If anyone is interested in helping with this issue please let me know.

### PR DESCRIPTION
**Pull request by tinsaye**
Closes #354 

The yarn script  `lint:i18n` uses the node package `i18n-unsued` to find and remove unused i18n keys. The default configuration of `i18n-unsued` does not find all usages, for example

- ` i18nKey="key"`
- `{t("key")}`
- `xyz.error(t("key"))`

Therefore, the config value `translationKeyMatcher` in `i18n-unused.config.js` specifies a custom regex (based on the [default](https://github.com/mxmvshnvsk/i18n-unused/blob/e522e2d737249ca9c939cdb35517be637e09b533/src/core/initialize.ts#L17)) to match the usages found in this source code.

### CODE REVIEW

**Comment by tinsaye** [comment]
I could also include the changed locals files after running the script if desired.

**Comment by lukevella** [comment]
Thanks @tinsaye. I just changed the base branch to development since there's no need to create a new production build for this. I'll test this out tomorrow.

>  I could also include the changed locals files after running the script if desired.

Not necessary, but feel free to push them if you have them.

File: package.json [hunk 1]
```diff
@@ -77,6 +78,7 @@
     "eslint-plugin-react": "^7.23.2",
     "eslint-plugin-react-hooks": "^4.2.0",
     "eslint-plugin-simple-import-sort": "^7.0.0",
+    "i18n-unused": "^0.12.0",
```
[the remark this answers is not in the document]
Comment (tinsaye) [reply package.json:81]: It is already :)

[the remark this answers is not in the document]
Comment (lukevella) [reply package.json:81]: Ah right, sorry it appeared as if it was under the other dependencies in the diff without expanding the file.

File: public/locales/en/app.json [hunk 2]
```diff
@@ -85,9 +83,7 @@
   "notificationsVerifyEmail": "You need to verify your email to turn on notifications",
   "noVotes": "No one has voted for this option",
   "ok": "Ok",
-  "options": "Options",
   "participant": "Participant",
-  "participantCount_other": "{{count}} participants",
```
Comment (lukevella) [inline public/locales/en/app.json:90]: So, we deal with [plurals](https://www.i18next.com/translation-function/plurals) by having these keys that are never used directly in the code but are still necessary. I believe i18n-unused can handle this ((see here)[https://github.com/mxmvshnvsk/i18n-unused/issues/13]) out of the box so not sure yet why these are being flagged 🤔

Comment (tinsaye) [reply public/locales/en/app.json:90]: Ah ok, then that was totally my mistake. I was searching for usages of [context](https://www.i18next.com/translation-function/context), but did not think of plurals. So I switched off the corresponding i18n-unused config just for one time. In hindsight, it should have been obvious. Will revert the last commit.

**Comment by lukevella** [comment]
Looks good to me. Thanks for your help @tinsaye!

**Pull request merged**